scripts/run_all.py

```diff
@@ -4,86 +4,239 @@
 
 
-def start_viewer(args: List[str] = []) -> Any:
-    process = subprocess.Popen(
-        ["cargo", "run", "-p", "rerun", "--all-features", "--"] + args,
-        stdout=subprocess.PIPE,
-    )
-    time.sleep(1)  # give it a moment to start
-    return process
+def collect_examples(fast: bool) -> List[str]:
+    if not fast:
+        return [os.path.dirname(entry) for entry in glob("examples/python/**/main.py")]
+    else:
+        # cherry picked
+        return [
+            "examples/python/deep_sdf",
+            "examples/python/dicom",
+            "examples/python/raw_mesh",
+            "examples/python/clock",
+            "examples/python/api_demo",
+            "examples/python/car",
+            "examples/python/colmap",
+            "examples/python/plots",
+            "examples/python/nyud",
+            "examples/python/text_logging",
+        ]
+
+
+class Viewer:
+    port: int
+    should_close: bool
+    web: bool
+    web_viewer_port: int
+    ws_server_port: int
+    process: Optional[Any]
+
+    def __init__(self, close: bool = False, web: bool = False):
+        self.port = get_free_port()
+        self.should_close = close
+        self.web = web
+        self.web_viewer_port = get_free_port()
+        self.ws_server_port = get_free_port()
+        self.process = None
+
+    def close(self) -> None:
+        if self.process is not None:
+            self.process.kill()
+        self.process = None
+
+    def start(self) -> "Viewer":
+        args = ["--port", str(self.port)]
+        if self.web:
+            args += [
+                "--web-viewer",
+                "--web-viewer-port",
+                str(self.web_viewer_port),
+                "--ws-server-port",
+                str(self.ws_server_port),
+            ]
+
+        self.process = subprocess.Popen(
+            ["cargo", "run", "-p", "rerun", "--all-features", "--"] + args,
+            stdout=subprocess.PIPE,
+        )
+        time.sleep(1)  # give it a moment to start
+        return self
+
+    def __enter__(self) -> "Viewer":
+        self.start()
+        return self
+
+    def __exit__(
+        self,
+        exc_type: Optional[Type[BaseException]],
+        exc: Optional[BaseException],
+        traceback: Optional[TracebackType],
+    ) -> None:
+        if self.process is not None and self.should_close:
+            self.process.kill()
+
+
+def run_sdk_build() -> None:
+    returncode = subprocess.Popen(
+        ["maturin", "develop", "--manifest-path", "rerun_py/Cargo.toml", '--extras="tests"'],
+    ).wait()
+    assert returncode == 0, f"process exited with error code {returncode}"
 
 
-def run_build() -> None:
-    process = subprocess.Popen(
-        ["maturin", "develop", "--manifest-path", "rerun_py/Cargo.toml", '--extras="tests"'],
-    )
-    returncode = process.wait()
+def run_viewer_build() -> None:
+    returncode = subprocess.Popen(["cargo", "build", "-p", "rerun", "--all-features"]).wait()
```

Comment:
Let's add the `--quite` flag to these to make the output of `run_all.py` more readable.